From aaed4d93b59aacd677d98b5ae790376a43db7df3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 22 Oct 2021 13:41:33 -0600
Subject: [PATCH] Lammps compatibility

---
 dynaphopy/interface/lammps_link.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/dynaphopy/interface/lammps_link.py b/dynaphopy/interface/lammps_link.py
index ebdd197..62df0c2 100644
--- a/dynaphopy/interface/lammps_link.py
+++ b/dynaphopy/interface/lammps_link.py
@@ -59,15 +59,15 @@ def generate_lammps_trajectory(structure,
     # print("Natoms, mass, x[0][0] coord =", natoms, mass[1], x[0][0])
     # print ('thermo', lmp.get_thermo('1'))
 
-    xlo =lmp.extract_global("boxxlo", 1)
-    xhi =lmp.extract_global("boxxhi", 1)
-    ylo =lmp.extract_global("boxylo", 1)
-    yhi =lmp.extract_global("boxyhi", 1)
-    zlo =lmp.extract_global("boxzlo", 1)
-    zhi =lmp.extract_global("boxzhi", 1)
-    xy =lmp.extract_global("xy", 1)
-    yz =lmp.extract_global("yz", 1)
-    xz =lmp.extract_global("xz", 1)
+    xlo =lmp.extract_global("boxxlo")
+    xhi =lmp.extract_global("boxxhi")
+    ylo =lmp.extract_global("boxylo")
+    yhi =lmp.extract_global("boxyhi")
+    zlo =lmp.extract_global("boxzlo")
+    zhi =lmp.extract_global("boxzhi")
+    xy =lmp.extract_global("xy")
+    yz =lmp.extract_global("yz")
+    xz =lmp.extract_global("xz")
 
     simulation_cell = np.array([[xhi-xlo, xy,  xz],
                            [0,  yhi-ylo,  yz],
