From 6c3553fd8281637dd67f59ba03a69c2d3be5fbc3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sat, 3 Jan 2026 22:09:07 +0100
Subject: [PATCH] Patch phonopy compatibility

---
 dynaphopy/interface/phonopy_link.py | 28 ++++++++++------------------
 1 file changed, 10 insertions(+), 18 deletions(-)

diff --git a/dynaphopy/interface/phonopy_link.py b/dynaphopy/interface/phonopy_link.py
index 5884625..ec2e6a2 100644
--- a/dynaphopy/interface/phonopy_link.py
+++ b/dynaphopy/interface/phonopy_link.py
@@ -135,11 +135,11 @@ def get_phonon(structure, NAC=False, setup_forces=True, custom_supercell=None, s
 
     if setup_forces:
         if structure.get_force_constants() is not None:
-            phonon.set_force_constants(structure.get_force_constants().get_array())
+            phonon.force_constants = structure.get_force_constants().get_array()
         elif structure.get_force_sets() is not None:
             phonon.set_displacement_dataset(structure.get_force_sets().get_dict())
             phonon.produce_force_constants()
-            structure.set_force_constants(ForceConstants(phonon.get_force_constants(),
+            structure.set_force_constants(ForceConstants(phonon.force_constants,
                                                          supercell=structure.get_force_sets().get_supercell()))
         else:
             print('No force sets/constants available!')
@@ -196,7 +196,7 @@ def obtain_phonopy_dos(structure, mesh=(40, 40, 40), force_constants=None,
                             setup_forces=False,
                             custom_supercell=force_constants.get_supercell(),
                             NAC=NAC)
-        phonon.set_force_constants(force_constants.get_array())
+        phonon.force_constants = force_constants.get_array()
 
     if projected_on_atom < 0:
         phonon.run_mesh(mesh)
@@ -233,7 +233,7 @@ def obtain_phonopy_thermal_properties(structure, temperature, mesh=(40, 40, 40),
                             setup_forces=False,
                             custom_supercell=force_constants.get_supercell(),
                             NAC=NAC)
-        phonon.set_force_constants(force_constants.get_array())
+        phonon.force_constants = force_constants.get_array()
 
     phonon.run_mesh(mesh)
     phonon.run_thermal_properties(t_step=1, t_min=temperature, t_max=temperature)
@@ -264,7 +264,7 @@ def obtain_phonopy_group_velocity(structure, q_point, force_constants=None, NAC=
                             setup_forces=False,
                             custom_supercell=force_constants.get_supercell(),
                             NAC=NAC)
-        phonon.set_force_constants(force_constants.get_array())
+        phonon.force_constants = force_constants.get_array()
 
     return phonon.get_group_velocity_at_q(q_point)
 
@@ -275,7 +275,7 @@ def obtain_phonopy_mesh_from_force_constants(structure, force_constants, mesh=(4
                         setup_forces=False,
                         custom_supercell=force_constants.get_supercell(),
                         NAC=NAC)
-    phonon.set_force_constants(force_constants.get_array())
+    phonon.force_constants = force_constants.get_array()
 
     phonon.run_mesh(mesh)
     mesh_dict = phonon.get_mesh_dict()
@@ -291,7 +291,7 @@ def obtain_phonon_dispersion_bands(structure, bands_ranges, force_constants=None
         phonon = get_phonon(structure, NAC=NAC, setup_forces=False,
                             custom_supercell=force_constants.get_supercell())
 
-        phonon.set_force_constants(force_constants.get_array())
+        phonon.force_constants = force_constants.get_array()
     else:
         # print('Getting phonon dispersion relations')
         phonon = get_phonon(structure, NAC=NAC)
@@ -315,11 +315,7 @@ def obtain_phonon_dispersion_bands(structure, bands_ranges, force_constants=None
 def get_commensurate_points(structure, fc_supercell):
 
     phonon = get_phonon(structure, setup_forces=False, custom_supercell=fc_supercell)
-
-    primitive = phonon.get_primitive()
-    supercell = phonon.get_supercell()
-
-    dynmat2fc = DynmatToForceConstants(primitive, supercell)
+    dynmat2fc = DynmatToForceConstants(phonon.primitive, phonon.supercell)
     com_points = dynmat2fc.get_commensurate_points()
 
     return com_points
@@ -335,7 +331,7 @@ def get_equivalent_q_points_by_symmetry(q_point, structure, symprec=None):
                         cell=structure.get_cell())
 
     tot_points = [list(q_point)]
-    for operation_matrix in Symmetry(bulk, symprec=symprec).get_reciprocal_operations():
+    for operation_matrix in Symmetry(bulk, symprec=symprec).reciprocal_operations:
         operation_matrix_q = np.dot(np.linalg.inv(structure.get_primitive_matrix()), operation_matrix.T)
         operation_matrix_q = np.dot(operation_matrix_q, structure.get_primitive_matrix())
 
@@ -351,11 +347,7 @@ def get_equivalent_q_points_by_symmetry(q_point, structure, symprec=None):
 def get_renormalized_force_constants(renormalized_frequencies, eigenvectors, structure, fc_supercell, symmetrize=False):
 
     phonon = get_phonon(structure, setup_forces=False, custom_supercell=fc_supercell)
-
-    primitive = phonon.get_primitive()
-    supercell = phonon.get_supercell()
-
-    dynmat2fc = DynmatToForceConstants(primitive, supercell)
+    dynmat2fc = DynmatToForceConstants(phonon.primitive, phonon.supercell)
 
     size = structure.get_number_of_dimensions() * structure.get_number_of_primitive_atoms()
     eigenvectors = np.array([eigenvector.reshape(size, size, order='C').T for eigenvector in eigenvectors])
