From c38a0a0bc30c6b0a15a753e64d2ada9cef2170ed Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Tue, 23 Jul 2024 20:13:29 +0200
Subject: [PATCH] setup patch

---
 setup.py | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 4dae0c8..8625eb6 100644
--- a/setup.py
+++ b/setup.py
@@ -14,11 +14,10 @@
 
 
 def check_compiler():
-    import subprocess
-    output = subprocess.Popen(['gcc'], stderr=subprocess.PIPE).communicate()[1]
-    if b'clang' in output:
+    import platform
+    if 'Darwin' in platform.system():
         return 'clang'
-    if b'gcc' in output:
+    elif 'Linux' in platform.system():
         return 'gcc'
 
 
@@ -77,7 +76,7 @@ def get_version_number():
                'scripts/fitdata',
                'scripts/qha_extract',
                'scripts/rfc_calc'],
-      install_requires=['phonopy', 'numpy', 'scipy', 'matplotlib'] + (["windows-curses"] if sys.platform in ["win32", "cygwin"] else []),
+      install_requires=['phonopy', 'numpy', 'scipy', 'matplotlib'],
       license='MIT License',
       long_description=open('README.md').read(),
       long_description_content_type='text/markdown',
